sdk/azidentity/aad_identity_client.go

```diff
@@ -41,8 +43,10 @@ const (
 // AADIdentityClient provides the base for authenticating with Client Secret Credentials, Client Certificate Credentials
 // and Environment Credentials. This type initializes a default azcore.Pipeline and IdentityClientOptions.
 type aadIdentityClient struct {
-	options  TokenCredentialOptions
-	pipeline azcore.Pipeline
+	lock         sync.RWMutex
+	options      TokenCredentialOptions
+	pipeline     azcore.Pipeline
+	refreshToken string
```

Comment:
This field, and all the device code-related methods, should go on the `DeviceCodeCredential` type.  Then you won't need the lock.

Reply:
Or perhaps some other internal struct, maybe even a `deviceCodeBearerAuthorizer`?

Reply:
I agree if we only decide to use the refresh token for DeviceCodeCredential. I was looking into what information is returned when requesting a token for the other credentials and several other credentials also return a refresh token. I can make a list of the information returned by the other credential types, then we can all discuss what to do with the information that's returned.

Reply:
The refresh token will be specific to each request possibly as it's specific to the user which logs in using the device code, hence it should definitely be stored on the `DeviceCodeCredential` and not on the client which can presumably handle request for many different credential instances.

Reply:
The refresh token will be specific to each request possibly as it's specific to the user which logs in using the device code, hence it should definitely be stored on the DeviceCodeCredential and not on the client which can presumably handle request for many different credential instances.